x/wasm/keeper/keeper.go

```diff
@@ -804,7 +789,24 @@ func (k Keeper) setContractInfoExtension(ctx sdk.Context, contractAddr sdk.AccAd
 
 // handleContractResponse processes the contract response
 func (k *Keeper) handleContractResponse(ctx sdk.Context, contractAddr sdk.AccAddress, ibcPort string, res *wasmvmtypes.Response) ([]byte, error) {
-	return k.wasmVMResponseHandler.Handle(ctx, contractAddr, ibcPort, res.Submessages, res.Messages, res.Data)
+	return k.doHandleWasmVMResponse(ctx, contractAddr, ibcPort, res.Submessages, res.Messages, res.Attributes, res.Data)
+}
+
+// doHandleWasmVMResponse processes the contract response data by emitting events and sending sub-/messages.
+func (k *Keeper) doHandleWasmVMResponse(
+	ctx sdk.Context,
+	contractAddr sdk.AccAddress,
+	ibcPort string,
+	subMsg []wasmvmtypes.SubMsg,
+	msgs []wasmvmtypes.CosmosMsg,
+	attrs []wasmvmtypes.EventAttribute,
+	data []byte,
+) ([]byte, error) {
+	// emit all events from this contract itself
+	events, customBytesLen := types.ParseEvents(attrs, contractAddr)
+	ctx.GasMeter().ConsumeGas(k.eventAttributeCost*customBytesLen, "Custom contract event attributes")
```

Comment:
I still kind of like giving a "free tier" like first 100 bytes are free, and charging afterwards.
To encourage people to produce some info, but not ridiculous amounts.

Of course by 100 I mean, "user configurable setting"

Reply:
Free tier and costs per attribute require 2 new cost attributes. It is reasonable to come up with some more advanced pricing model than just bytes charge. I only see that my keeper attributes do not scale well nor provide enough flexibility for custom costs models. 
I will redesign this into a new component that provides defaults but can be substituted with custom logic as extension point